java/src/main/native/src/row_conversion.cu

```diff
@@ -1395,42 +1448,40 @@ struct column_info_s {
  */
 template <typename iterator>
 column_info_s compute_column_information(iterator begin, iterator end) {
-  size_type fixed_width_size_per_row = 0;
-  std::vector<size_type> fixed_width_column_starts;
-  std::vector<size_type> fixed_width_column_sizes;
+  size_type size_per_row = 0;
+  std::vector<size_type> column_starts;
+  std::vector<size_type> column_sizes;
   std::vector<size_type> variable_width_column_starts;
 
-  for (auto cv = begin; cv != end; ++cv) {
-    auto col_type = std::get<0>(*cv);
-    bool const compound_type = is_compound(col_type);
+  for (auto col_type = begin; col_type != end; ++col_type) {
+    bool const compound_type = is_compound(*col_type);
 
     // a list or string column will write a single uint64
     // of data here for offset/length
-    auto const col_size = compound_type ? sizeof(uint32_t) + sizeof(uint32_t) : size_of(col_type);
+    auto const col_size = compound_type ? sizeof(uint32_t) + sizeof(uint32_t) : size_of(*col_type);
```

Comment:
Why not just `sizeof(int64_t)`?

Reply:
The jcudf format specifies that this is a 8 bytes, but it is actually two int32's and requires 4 byte alignment. I just wanted to be explicit here that it isn't a single uint64.